### PR TITLE
Atk Canceller refactor / reorder / clean up

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Changes based loosely on [smogons reserch](https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/page-54#post-8548957)

Accidentally fixes a Stance Change bug

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
ready for review

File: test/battle/gimmick/dynamax.c [hunk 1]
```diff
@@ -1431,6 +1431,7 @@ DOUBLE_BATTLE_TEST("(DYNAMAX) G-Max One Blow bypasses Max Guard for full damage"
 }
 
 // Bug Testing
+//  This test will fail if it's the first test a thread runs
```
Comment (Bassoonian) [inline test/battle/gimmick/dynamax.c:1434]: Any idea why?

Comment (AlexOn1ine) [reply test/battle/gimmick/dynamax.c:1434]: Nope. There are two other comments like that in this file

Comment (AsparagusEduardo) [reply test/battle/gimmick/dynamax.c:1434]: Confirmed that it's a thing as of current master

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**